janis_unix/data_types/text.py

```diff
@@ -11,3 +15,66 @@ def name():
 
     def doc(self):
         return "A textfile, ending with .txt"
+
+    @classmethod
+    def basic_test(
+        cls,
+        tag: str,
+        min_size: int,
+        min_required_content: Optional[str] = None,
+        line_count: Optional[int] = None,
+        md5: Optional[str] = None,
+        expected_file_path: Optional[str] = None,
+    ) -> List[TTestExpectedOutput]:
+        outcome = [
+            TTestExpectedOutput(
+                tag=tag,
+                preprocessor=TTestPreprocessor.FileSize,
+                operator=operator.ge,
+                expected_value=min_size,
+            ),
+        ]
+
+        if min_required_content is not None:
+            outcome += [
+                TTestExpectedOutput(
+                    tag=tag,
+                    preprocessor=TTestPreprocessor.FileContent,
+                    operator=operator.contains,
+                    expected_value=min_required_content,
+                ),
+            ]
+
+        if line_count is not None:
+            outcome += [
+                TTestExpectedOutput(
+                    tag=tag,
+                    preprocessor=TTestPreprocessor.LineCount,
+                    operator=operator.eq,
+                    expected_value=line_count,
+                ),
+            ]
+
+        if md5 is not None:
+            outcome += [
+                TTestExpectedOutput(
+                    tag=tag,
+                    preprocessor=TTestPreprocessor.FileMd5,
+                    operator=operator.eq,
+                    expected_value=md5,
+                ),
+            ]
+
+        if expected_file_path is not None:
+            with open(expected_file_path, "r") as f:
+                expected_content = f.read()
+            outcome += [
+                TTestExpectedOutput(
+                    tag=tag,
+                    preprocessor=TTestPreprocessor.FileContent,
+                    operator=operator.eq,
+                    expected_value=expected_content,
```

Comment:
@xinzhel97 I know this has been merged. But, can you create another pull request and change this so that we use the existing field.
```
expected_file=expected_file_path
```
If we use this, it will open the file content at the time we perform the assertion.

and we remove these lines:
```
with open(expected_file_path, "r") as f:
                expected_content = f.read()
```
Thanks!

Reply:
Sorry! I must have overlooked this message . I have just created another pull request per instruction #10